src/lib/transaction_snark/test/zkapps_examples/verification_key_modify/verification_key_modify.ml

```diff
@@ -0,0 +1,277 @@
+        let account_updates_hash =
+          Zkapp_command.Call_forest.hash zkapp_command
+        in
+        Zkapp_command.Transaction_commitment.create ~account_updates_hash
+      in
+      let fee_payer : Account_update.Fee_payer.t =
+        { body =
+            { Account_update.Body.Fee_payer.dummy with
+              public_key = pk_compressed
+            ; fee = Currency.Fee.(of_nanomina_int_exn 100)
+            }
+        ; authorization = Signature.dummy
+        }
+      in
+      let memo_hash = Signed_command_memo.hash memo in
+      let full_commitment =
+        Zkapp_command.Transaction_commitment.create_complete
+          transaction_commitment ~memo_hash
+          ~fee_payer_hash:
+            (Zkapp_command.Call_forest.Digest.Account_update.create
+               (Account_update.of_fee_payer fee_payer) )
+      in
+      let sign_all ({ fee_payer; account_updates; memo } : Zkapp_command.t) :
+          Zkapp_command.t =
+        let fee_payer =
+          match fee_payer with
+          | { body = { public_key; _ }; _ }
+            when Public_key.Compressed.equal public_key pk_compressed ->
+              { fee_payer with
+                authorization =
+                  Schnorr.Chunked.sign sk
+                    (Random_oracle.Input.Chunked.field full_commitment)
+              }
+          | fee_payer ->
+              fee_payer
+        in
+        let account_updates =
+          Zkapp_command.Call_forest.map account_updates ~f:(function
+            | ({ body = { public_key; use_full_commitment; _ }
+               ; authorization = Signature _
+               } as account_update :
+                Account_update.t )
+              when Public_key.Compressed.equal public_key pk_compressed ->
+                let commitment =
+                  if use_full_commitment then full_commitment
+                  else transaction_commitment
+                in
+                { account_update with
+                  authorization =
+                    Signature
+                      (Schnorr.Chunked.sign sk
+                         (Random_oracle.Input.Chunked.field commitment) )
+                }
+            | account_update ->
+                account_update )
+        in
+        { fee_payer; account_updates; memo }
+      in
+      let zkapp_command : Zkapp_command.t =
+        sign_all { fee_payer; account_updates = zkapp_command; memo }
+      in
+      Ledger.with_ledger ~depth:ledger_depth ~f:(fun ledger ->
+          let account =
+            Account.create account_id
+              Currency.Balance.(
+                Option.value_exn
+                  (add_amount zero (Currency.Amount.of_nanomina_int_exn 500)))
+          in
+          let _, loc =
+            Ledger.get_or_create_account ledger account_id account
+            |> Or_error.ok_exn
+          in
+          Async.Thread_safe.block_on_async_exn (fun () ->
+              check_zkapp_command_with_merges_exn ?expected_failure ledger
```

Comment:
Call the verifier as well, to do the outside snark verification? You could add a check like this https://github.com/MinaProtocol/mina/blob/a97e94375fb537651c7dc7c2c18e584748ad6cb3/src/app/cli/src/init/transaction_snark_profiler.ml#L610 to `check_zkapp_command_with_merges_exn`